### PR TITLE
Update FC_Tests.xml to support multiple fc adapters

### PR DESCRIPTION
**Pull request by lixuemin2016**
Based on existing FC two test cases, FC_WWN_Basic and FC_Disks_Detection to add two new test cases, including FC_WWN_Basic_Multi_Adapters and FC_Disks_Detection_Multi_Adapters.

1.  FC_Disks_Detection_Multi_Adapters -- since the vm can add 4 FC adapters as maximum number, locally we have case to cover this test point, add new case here.
2.   FC_WWN_Basic_Multi_Adapters -- this is based on bug https://bugzilla.redhat.com/show_bug.cgi?id=1308632,  check /sys/class/fc_host/hostx,  already have auto case FC_WWN_Basic to cover one adapter, locally we have test case to cover multiple adapters. 

Any comment or suggestion, will update.

Thank you so much.
Best Regards,
Xuemin

### CODE REVIEW

**Comment by ampintilie** [comment]
Hello, I saw in the logs this message "/root/FC_disks.sh: line 187: CheckCallTracesWithDelay: command not found". As far as I've seen, the utils.sh is not sourced yet in FC_disk.sh. Also, the xml needs to be updated.

Thanks,
Ana

File: WS2012R2/lisa/remote-scripts/ica/FC_disks.sh [hunk 1]
```diff
@@ -182,6 +182,10 @@ do
         UpdateTestState $ICA_TESTFAILED
         exit 90
     fi
+
+# Check for Call traces
+CheckCallTracesWithDelay 20
```
Comment (chvalean) [inline WS2012R2/lisa/remote-scripts/ica/FC_disks.sh]: must source utils.sh in order to call this function

Comment (lixuemin2016) [reply WS2012R2/lisa/remote-scripts/ica/FC_disks.sh]: yes, sorry, my mistake, thank you.

**Comment by lixuemin2016** [comment]
Thank you ampintilie help to review, will submit change soon.

**Comment by lixuemin2016** [comment]
@ampintilie @chvalean, updated source utils.sh and related xml  based on comment, thank you so much for your review.

File: WS2012R2/lisa/xml/FC_Tests.xml [hunk 2]
```diff
@@ -105,6 +111,47 @@
...
+		<test>
+			<testName>FC_WWN_Basic_Multi_Adapters</testName>
+			<testScript>setupScripts\FC_WWN.ps1</testScript>
+			<files>remote-scripts\ica\fc_wwn.sh,remote-scripts/ica/utils.sh</files>
+			<setupScript>setupscripts\FC_AddFibreChannelHba.ps1</setupScript>
+			<cleanupScript>setupScripts\FC_RemoveFibreChannelHba.ps1</cleanupScript>
+			<timeout>800</timeout>
+			<testParams>
+				<param>TC_COVERED=FC-11</param>
+				<param>WWNN1=C003FFCBF9850025</param>
```
[the remark this answers is not in the document]
Comment (lixuemin2016) [reply WS2012R2/lisa/xml/FC_Tests.xml]: Yes, current only support one SAN, would you like to support more SANs? 

Here multiple adapters means multiple "Fibre Channel Adapter" in VM.

 In the Hyper-v manager, selecting "Add Hardware" -> ""Fibre Channel Adapter" -> "Add",  select   "Virtual SAN", add Port addresses to add "Fibre Channel Adapter", e.g "C003FFCBF9850021"，"C003FFCBF98500xx" connected with disk.  We can add maximum 4 "Fibre Channel Adapter" with same "Virtual SAN" or different "Virtual SAN" name. 

If use same "Virtual SAN"(configure one or more physical WWPN) in the Virtual Fibre Channel SANs),  we have 4 "different virtual ports (virtual WWPN)" which are connected with storage on the different "Fibre Channel Adapter", then can detect multiple disks in the VM  even on the same SAN.  Is it reasonable?

Notice that in the FC_Tests.xml already have one "WWNN=", so in the test case, add WWNN1 to WWNN3 to param.

Thank you so much.

[the remark this answers is not in the document]
Comment (chvalean) [reply WS2012R2/lisa/xml/FC_Tests.xml]: Thank you for the explanation, this makes sense then.

**Comment by ampintilie** [comment]
Hello @lixuemin2016 , 
In case of FC_Disks_Detection_Multi_Adapters, there are different disks for each fibre channel adapter? In this situation we don't have to check the read/write operation on all disks? Please tell me if I'm wrong.
Thank you.

**Comment by lixuemin2016** [comment]
Hello @ampintilie , thank you so much for your reminder, yes, each fibre channel adapter has different disk. We'd better check all the read/write operation on all disks, will add new pull request for it. (I'm not sure original author's purpose about only checking one disk, and adding comment about how to testing multiple disks in the description part )
Here we cannot get the exact expected disk numbers, if add virtual port address without disk connected(fake WWPN), it does not have disk. As long as one valid WWPN in the multiple FC adapters, this case can test pass. 

Thank you.

**Pull request merged**